### ISSUE TITLE
code format: use dotnet format, add .editorconfig, add to CI

### ISSUE THREAD

**Issue by nietras** [CarlVerret/csFastFloat#101]
It would be nice if the code followed "normal" C# formatting and simply used "dotnet format" as part of workflow and CI. Since I use format on save I had to do quite a bit of jumping back and forth and disable it when doing the #100 PR.

I could potentially adopt practices and workflows from https://github.com/nietras/Sep here if there was an interest in that.

**Comment by CarlVerret** [comment]
I totally agree.  I did a lot of changes to the code formating when I merged our project into .net 7 std library.  I should have did the same to this lib a while ago.
I'm not really familiar about how to add the dotnet format in the CI, I'd be glad to learn about it if you've got any info about that.   In the meantime, i just merged a new version that includes an editorconfig file !

**Comment by nietras** [comment]
great, I believe it is best if editorconfig is in root of project.
